internal/types/expression.go

```diff
@@ -68,15 +68,25 @@ func (e *FieldPathError) Code() ExpressionErrorCode {
 // Expression is an expression constructed from field value.
 type Expression interface {
 	Evaluate(doc *Document) any
+	GetPath() Path
 }
 
 // pathExpression is field path constructed from expression.
 type pathExpression struct {
 	path Path
+	*ExpressionOpts
 }
 
-// NewExpression creates a new instance by checking expression string.
-func NewExpression(expression string) (Expression, error) {
+// ExpressionOpts represents options used to modify behavior of Expression functions.
+type ExpressionOpts struct {
+	// IgnoreArrays disables checking arrays for provided key.
+	// So expression {"$v.foo"} won't match {"v":[{"foo":42}]}
+	IgnoreArrays bool // defaults to false
```

Comment:
Many way of getting path :sweat_smile:  Thanks for documenting this.

Reply:
Part of `getExpressionPathValue` has similarity with `getDocumentsAtSuffix` see doc string of https://github.com/FerretDB/FerretDB/pull/2294/files#diff-370ad7a00541d3a0aa2e121e3232791b1b2b3c432ee4f895917a48dec6714a76R196.

I'm starting to think that, we might actually be better off to merge `getExpressionPathValue` and `getDocumentsAtSuffix` then have options like here to decide what to match. Out side of the scope of this work.

Reply:
That's a great idea. I will create an issue to track that. Maybe we'll find more options to apply there.

Reply:
Done https://github.com/FerretDB/FerretDB/issues/2348